harness/features/track.local.test.ts

```diff
@@ -0,0 +1,355 @@
+import {
+    getConnectionStringForProxy,
+    forEachSDK,
+    describeIf,
+    createClient,
+    createUser,
+    wait,
+    callOnClientInitialized,
+    callTrack
+} from '../helpers'
+import { Capabilities, SDKCapabilities } from '../types'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../nock'
+import { config } from '../mockData/index'
+import nock from 'nock'
+
+jest.setTimeout(15000)
+
+const scope = getServerScope()
+
+describe('Track Tests - Local', () => {
+    const validUserId = 'user1'
+    forEachSDK((name) => {
+        let url: string
+        const capabilities: string[] = SDKCapabilities[name]
+        const clientId: string = uuidv4()
+        const mockServerUrl
+            = `http://${process.env.DOCKER_HOST_IP ?? 'host.docker.internal'}:${global.__MOCK_SERVER_PORT__}`
+        const eventFlushIntervalMS = 1000
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)
+
+            const sdkKey = `dvc_server_${clientId}`
+
+            scope
+                .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                .reply(200, config)
+
+            scope.post(`/client/${clientId}`).reply((uri, body) => {
+                if (typeof body === 'object'
+                    && body.message.includes('onClientInitialized was invoked'))
+                    return [200]
+            })
+
+            await createClient(url, clientId, sdkKey, {
+                baseURLOverride: `${mockServerUrl}/client/${clientId}`,
+                eventFlushIntervalMS: eventFlushIntervalMS, logLevel: 'debug', configPollingIntervalMS: 1000 * 60
+            })
+            await callOnClientInitialized(clientId, url, `${mockServerUrl}/client/${clientId}`)
+
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            describe('Expect no events sent', () => {
+                it('should not send an event if the event type not set', async () => {
+                    let eventBody = {}
+
+                    const response = await createUser(url, { user_id: validUserId })
+                    await response.json()
+                    const userId = response.headers.get('location')
+
+                    const trackResponse = await callTrack(clientId, url, userId, {})
+
+                    nock.emitter.on('no match', (req) => {
```

Comment:
if this is testing that no events are sent then this should never happen

Reply:
oh yes gets caught globally

Reply:
yeah but it shouldn't hit an unmatched request in this test, since no flush request should be made (since the event is invalid)